### ISSUE TITLE
jbuilder build fails in file "test.ml"

### ISSUE THREAD

**Issue by colinxy** [onionhoney/stlc-ocaml#1]
After installing all the dependencies, with `opam install core ppx_jane menhir` and `jbuilder external-lib-deps --missing test.exe`, 

```
$ jbuilder build test.exe
      menhir stlc.{ml,mli}
Warning: 6 states have shift/reduce conflicts.
Warning: one state has reduce/reduce conflicts.
Warning: 42 shift/reduce conflicts were arbitrarily resolved.
Warning: 9 reduce/reduce conflicts were arbitrarily resolved.
      ocamlc .test.eobjs/test.{cmi,cmo,cmt} (exit 2)
(cd _build/default && /usr/bin/ocamlc.opt -w -40 -g -bin-annot -I /home/colinxy/.opam/default/lib/base -I /home/colinxy/.opam/default/lib/base/caml -I /home/colinxy/.opam/default/lib/base/md5 -I /home/colinxy/.opam/default/lib/base/shadow_stdlib -I /home/colinxy/.opam/default/lib/bin_prot -I /home/colinxy/.opam/default/lib/bin_prot/shape -I /home/colinxy/.opam/default/lib/core -I /home/colinxy/.opam/default/lib/core_kernel -I /home/colinxy/.opam/default/lib/core_kernel/base_for_tests -I /home/colinxy/.opam/default/lib/fieldslib -I /home/colinxy/.opam/default/lib/jane-street-headers -I /home/colinxy/.opam/default/lib/ppx_assert/runtime-lib -I /home/colinxy/.opam/default/lib/ppx_bench/runtime-lib -I /home/colinxy/.opam/default/lib/ppx_compare/runtime-lib -I /home/colinxy/.opam/default/lib/ppx_expect/collector -I /home/colinxy/.opam/default/lib/ppx_expect/common -I /home/colinxy/.opam/default/lib/ppx_expect/config -I /home/colinxy/.opam/default/lib/ppx_hash/runtime-lib -I /home/colinxy/.opam/default/lib/ppx_inline_test/config -I /home/colinxy/.opam/default/lib/ppx_inline_test/runtime-lib -I /home/colinxy/.opam/default/lib/sexplib -I /home/colinxy/.opam/default/lib/sexplib/0 -I /home/colinxy/.opam/default/lib/sexplib/unix -I /home/colinxy/.opam/default/lib/spawn -I /home/colinxy/.opam/default/lib/stdio -I /home/colinxy/.opam/default/lib/typerep -I /home/colinxy/.opam/default/lib/variantslib -I /usr/lib64/ocaml/threads -no-alias-deps -I .test.eobjs -o .test.eobjs/test.cmo -c -impl test.pp.ml)
File "test.ml", line 63, characters 4-8:
Error: The function applied to this argument has type
         ?readme:(unit -> string) -> Core__Command.t
This argument cannot be applied without label
```

Related version info
```
$ uname -a
Linux *ip redacted* 4.15.8-300.fc27.x86_64 #1 SMP Fri Mar 9 18:11:36 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
$ ocaml --version
The OCaml toplevel, version 4.05.0
$ opam --version
2.0.0~beta6
```

**Comment by onionhoney** [comment]
Can you try `opam update; opam upgrade` ? 
When I upgraded OCaml from 4.03.0 to 4.06.1 there was a dependency issue too, which is fixed by upgrading all the opam packages.

**Comment by colinxy** [comment]
Still the same error after upgrade.

**Comment by colinxy** [comment]
Here are relevant packages and their versions.

```
$ opam list | grep -e core -e ppx
core                    v0.10.0      Industrial strength alternative to OCaml's standard library
core_kernel             v0.10.0      Industrial strength alternative to OCaml's standard library
ppx_assert              v0.10.0      Assert-like extension nodes that raise useful errors on failure
ppx_ast                 v0.10.0      OCaml AST used by Jane Street ppx rewriters
ppx_base                v0.10.0      Base set of ppx rewriters
ppx_bench               v0.10.0      Syntax extension for writing in-line benchmarks in ocaml code
ppx_bin_prot            v0.10.0      Generation of bin_prot readers and writers from types
ppx_compare             v0.10.0      Generation of comparison functions from types
ppx_core                v0.10.0      Standard library for ppx rewriters
ppx_custom_printf       v0.10.0      Printf-style format-strings for user-defined string conversion
ppx_derivers            1.0          Shared [@@deriving] plugin registry
ppx_driver              v0.10.3      Feature-full driver for OCaml AST transformers
ppx_enumerate           v0.10.0      Generate a list containing all values of a finite type
ppx_expect              v0.10.1      Cram like framework for OCaml
ppx_fail                v0.10.0      Add location to calls to failwiths
ppx_fields_conv         v0.10.0      Generation of accessor and iteration functions for ocaml records
ppx_hash                v0.10.0      A ppx rewriter that generates hash functions from type expressions and definitions
ppx_here                v0.10.0      Expands [%here] into its location
ppx_inline_test         v0.10.1      Syntax extension for writing in-line tests in ocaml code
ppx_jane                v0.10.0      Standard Jane Street ppx rewriters
ppx_js_style            v0.10.0      Code style checker for Jane Street Packages
ppx_let                 v0.10.0      Monadic let-bindings
ppx_metaquot            v0.10.0      Write OCaml AST fragment using OCaml syntax
ppx_optcomp             v0.10.0      Optional compilation for OCaml
ppx_optional            v0.10.0      Pattern matching on flat options
ppx_pipebang            v0.10.0      A ppx rewriter that inlines reverse application operators `|>` and `|!`
ppx_sexp_conv           v0.10.0      Generation of S-expression conversion functions from type definitions
ppx_sexp_message        v0.10.0      A ppx rewriter for easy construction of s-expressions
ppx_sexp_value          v0.10.0      A ppx rewriter that simplifies building s-expressions from ocaml values
ppx_tools_versioned     5.1          A variant of ppx_tools based on ocaml-migrate-parsetree
ppx_traverse            v0.10.0      Automatic generation of open-recursion classes
ppx_traverse_builtins   v0.10.0      Builtins for Ppx_traverse
ppx_type_conv           v0.10.0      Support Library for type-driven code generators
ppx_typerep_conv        v0.10.0      Generation of runtime types from type declarations
ppx_variants_conv       v0.10.0      Generation of accessor and iteration functions for ocaml variant types
```

**Comment by onionhoney** [comment]
Okay, so I investigated into v0.10 of Core. It turns out they plan to deprecate Command.spec and so the old Command.basic has been renamed to Command.basic_spec. I will fix this. 
Edit : fixed in 07259817687b372d4ced9686123f56de9d70d95e.
